test/common/testlib.py

```diff
@@ -1276,8 +1276,12 @@ def kill_global_machine(cls):
     def label(self):
         return self.__class__.__name__ + '-' + self._testMethodName
 
-    def new_machine(self, image=None, forward=None, restrict=True, cleanup=True, **kwargs):
-        machine_class = self.machine_class
+    def new_machine(self, image=None, forward=None, restrict=True, cleanup=True, inherit_machine_class=True, **kwargs):
+        if inherit_machine_class:
+            machine_class = inherit_machine_class and self.machine_class or testvm.VirtMachine
+        else:
+            machine_class = testvm.VirtMachine
```

Comment:
Please drop the `if:` and `else:` branch. Line 1281 is sufficient to cover all cases.

Reply:
oops thanks.